### ISSUE TITLE
Packer Vagrant Builder no longer working in 1.4.3

### ISSUE THREAD

**Issue by dasginganinja** [hashicorp/packer#8019]
Packer Version: 1.4.3 (from brew)
Host Platform: OSX 10.14.16
Virtualbox 6.0.10

Log Output / Template:
https://gist.github.com/dasginganinja/2c77432e6e18cc9627e0f697ee85df66

Description:
I recently upgraded to packer 1.4.3 from 1.4.1. When running the following command the behavior has changed and now vagrant complains about a missing `packer_vagrant` file that I have not referenced anywhere. 

`Couldn't open file <path>/packer_vagrant`

Current Workaround:
Uninstalling 1.4.3 and rolling back to 1.4.1 exhibits previously functioning behavior.

**Comment by phpock** [comment]
Looks like the box_name (which deaults to "packer_" plus your buildname) is used as the source box name in the generated Vagrantfile. If you don't want to add the box anyhow (i.e. skip_add = true), you can specify the name of the source box as box_name.

For example, based on the Gist above:
```json
{
  "variables": {
    "box_version": "1.4-dev",
    "source_box_name": "generic/debian9",
    "dest_box_name": "LehighWMS-StretchLamp"
  },
  "builders": [
    {
      "box_name": "{{user `source_box_name`}}",
      "communicator": "ssh",
      "source_path": "{{user `source_box_name`}}",
      "provider": "virtualbox",
      "skip_add": true,
      "type": "vagrant",
      "output_dir": "build/{{user `dest_box_name`}}"
    }
  ]
}
```

**Comment by angusfz** [comment]
+1 For the same issue (packer 1.4.3@mac)

**Comment by SwampDragons** [comment]
I think this is a duplicate of #7667; please check that issue out for updates and workarounds.

**Comment by SwampDragons** [comment]
Oops, commented on and closed wrong issue 🤦‍♀

**Comment by SwampDragons** [comment]
I believe this should be fixed by PR #8125. 

You can find builds of the patch here: https://circleci.com/gh/hashicorp/packer/12898#artifacts/containers/0

If any of y'all want to verify for me that your Vagrant builder is working properly now, I'd appreciate it :)

**Comment by tomasbackman** [comment]
I was also bitten by the orignal problem using 1.4.3, using this builder:
``` ... {
      "type": "vagrant",
      "name": "myCoolBox",
      "source_path": "ubuntu/bionic64",
      "communicator": "ssh",
      "skip_add": true,
      "template": "templates/Vagrantfile",
    }```
And this in the Vagrantfile: 
``` Vagrant.configure("2") do |config|
    config.vm.box = "{{.BoxName}}" ```

 
Like suggested it was solved by adding `"box_name": "ubuntu/bionic64"` since it otherwise defaulted to `packer_myCoolBox` and then could not find that box..

When testing/running I have found out that this configuration works for version 1.4.3 (and 1.42 and 1.4.1) BUT NOT with 1.4.4.. nor latest 1.5.0-Dev.
So something has been changed again in the behaviour, so maybe this should be reopened? (or at least investigated? @SwampDragons )
